nexus/db-queries/src/db/datastore/disk.rs

```diff
@@ -336,6 +336,8 @@ impl DataStore {
                 .filter(instance::dsl::state
                         .eq_any(ok_to_detach_instance_states)
                         .and(instance::dsl::active_propolis_id.is_null())),
+                        // TODO: sure did think i could just call .name() here...
+//                        .and(instance::dsl::boot_device.ne(authz_disk.name()))),
```

Comment:
i'd _thought_ that asserting a to-be-detached disk is not the boot device would be this simple. it is not? i've not figured out what an `authz::*` is vs a `db::model::*` yet, but i suspect there's something close to what i've written that does the job

Reply:
Storing the ID instead solves this, right?

Reply:
i think so!

Reply:
it turns out `authz_thing` is not "wrapper for `thing` that has access control associated with it", it's more like "access control for a resource, and the ID of the referenced resource". there was no name here in the first place. if i'd wanted to use the device name here, i'd have had to look up the disk by ID and use that.

now i know!